### PR TITLE
Grpc fix response types for batches to support LWTs

### PR DESCRIPTION
**Pull request by tomekl007**

### CODE REVIEW

File: grpc/src/main/java/io/stargate/grpc/service/Service.java [hunk 1]
```diff
@@ -644,9 +645,18 @@ private ConsistencyLevel getTracingConsistency(BatchParameters parameters) {
...
-          if (result.kind != Kind.Void) {
+
+          if (result.kind != Kind.Void && result.kind != Kind.Rows) {
             throw Status.INTERNAL.withDescription("Unhandled result kind").asException();
           }
+
+          if (result.kind == Kind.Rows) {
+            Payload.Type type = validateAndGetTypeForBatchQueries(batch);
+            PayloadHandler handler = PayloadHandlers.get(type);
+            Any data = handler.processResult((Rows) result, false);
```
Comment (mpenick) [inline grpc/src/main/java/io/stargate/grpc/service/Service.java:656]: Should skipping the metadata be a parameter exposed to the user? (maybe in `Batch`) I could go either way here.

File: grpc/src/main/java/io/stargate/grpc/service/Service.java [hunk 2]
```diff
@@ -644,9 +645,18 @@ private ConsistencyLevel getTracingConsistency(BatchParameters parameters) {
...
           Response.Builder responseBuilder = makeResponseBuilder(result);
           handleTraceId(result.getTracingId(), parameters, responseAndTraceId);
-          if (result.kind != Kind.Void) {
+
+          if (result.kind != Kind.Void && result.kind != Kind.Rows) {
             throw Status.INTERNAL.withDescription("Unhandled result kind").asException();
           }
+
+          if (result.kind == Kind.Rows) {
+            Payload.Type type = validateAndGetTypeForBatchQueries(batch);
```
Comment (mpenick) [inline grpc/src/main/java/io/stargate/grpc/service/Service.java:654]: Maybe type validation could be done earlier (before the queries are executed)?

File: grpc/src/main/java/io/stargate/grpc/service/Service.java (hunk 1 above)
Comment (tomekl007) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java]: good point, I've added a version of method that takes `BatchParameters`

File: grpc/src/main/java/io/stargate/grpc/service/Service.java (hunk 2 above)
Comment (tomekl007) [reply grpc/src/main/java/io/stargate/grpc/service/Service.java:654]: done

**Review by mpenick — APPROVED** [review]

**Pull request merged**